### ISSUE TITLE
lag?

### ISSUE THREAD

**Issue by 90th** [refact0r/system24#2]
why is it that when i have this theme on and i type fast it seems to lag lol?

**Comment by 90th** [comment]
i removed the theme and it seemed to fix it 🤔

**Comment by 90th** [comment]
https://i.imgur.com/JJj2yI8.png

reading names in VC's is also pretty horrible.

**Comment by refact0r** [comment]
> reading names in VC's is also pretty horrible.

i'm aware, there are a lot of small places that need to be patched because discord sucks

**Comment by refact0r** [comment]
> why is it that when i have this theme on and i type fast it seems to lag lol?

how strong is your computer? themes will inevitably cause lag, especially attribute selector based themes like this one. i just need to know if there's something making it really bad. 

can you try my midnight theme and see if you still get lag? because if you get lag on that as well then its just your computer not being good enough.

**Comment by 90th** [comment]
GTX 1060 6gb
16gb ram
i5 8600

**Comment by refact0r** [comment]
> can you try my midnight theme and see if you still get lag? because if you get lag on that as well then its just your computer not being good enough.

please try this.

**Comment by 90th** [comment]
midnight theme feels smooth when typing fast

**Comment by refact0r** [comment]
alright i'll see what i can do, but to get same performance as midnight theme would require a complete rewrite of the theme.

**Comment by 90th** [comment]
The lag isn't anything crazy, just more of an OCD thing, and only happens when I'm typing extremely fasting.

**Comment by refact0r** [comment]
hey, can you check if performance has improved?

**Comment by LyingOnCables** [comment]
> hey, can you check if performance has improved?

Hey, I was having the same problem, when typing fast, get slow for some reason. 
With this update seems to be working like a charm now.
Btw, your theme is really beautiful!

**Comment by LyingOnCables** [comment]
If this helps in someway, it used to happen in a channel with a lot of text.
![Discord_yKRVUYTyFf](https://github.com/user-attachments/assets/30dc57f1-1ee6-47ff-892c-62c3b0181b4c)

Like this one:
![floorp_wr3x0U5FBS](https://github.com/user-attachments/assets/e0855865-24f6-450d-9898-ef80546c3632)

**Comment by refact0r** [comment]
great. if anyone still has lag problems, please comment, otherwise i will close this issue.

**Comment by blindmodz** [comment]
At least for me it's fixed

**Comment by 90th** [comment]
I dont even notice it anymore like i used to, so i would say its fixed

**Comment by kilcha** [comment]
I noticed that lags occur on my main computer:
Xeon E5 2650 V2
GTX 1660 Super
16 GB RAM
And on my laptop everything works smoothly:
HP Pavilion Plus Laptop 14-ey00
It has an AMD processor and integrated graphics. On the main computer it lags when typing text, but not on the laptop. Strange.

**Comment by ninetailedtori** [comment]
Yeah, there's a LOT of lag with discord channels that are filled up, and given most discord channels likely will be, this is a bit annoying :,)
Running Hyprland on Arch on Zen kernel, Ryzen 5950X and RTX 3080Ti with 32G DDR4-3603.

**Comment by Akasiek** [comment]
Please, reopen this issue. It's still relevant

**Comment by ninetailedtori** [comment]
I'm physically unable to use this theme as it gets so laggy to use that there's about 3 seconds of lag behind my mouse or text cursor. Which is sad, because it's very pretty

**Comment by refact0r** [comment]
please read my comments. it is not something simple i can fix. if you have a weak computer then don't use this theme, use midnight or another class-based theme.

**Comment by ninetailedtori** [comment]
I'm using an incredibly strong computer...which just makes this more depressing lmao

**Comment by refact0r** [comment]
> Yeah, there's a LOT of lag with discord channels that are filled up, and given most discord channels likely will be, this is a bit annoying :,) Running Hyprland on Arch on Zen kernel, Ryzen 5950X and RTX 3080Ti with 32G DDR4-3603, and vencord to test.

and this only happens with the theme enabled? there are a number of other things that could affect performance:

1. use vesktop instead of normal vencord or betterdiscord
2. check your plugins
3. try enabling/disabling graphics acceleration

also there was an update very recently so discord may have broke something related to clients, who knows. or it could be something related to arch/hyprland.

**Comment by ninetailedtori** [comment]
I'm using vesktop right now, I do have plugins but I've not got lag with them and no theme...graphics acceleration, well, without it the app just breaks. Settings menu is fine, but when I'm in a chat, or even in the friends list? Lagged to hell for some reason.

**Comment by ninetailedtori** [comment]
When using midnight, zero lag at all lmao, in fact it's smoother than the default experience. But sys24 is just not working on my system for some reason.

**Comment by refact0r** [comment]
can everyone in this thread check if performance has improved? i've added a style that supposedly improves scrolling performance in particular. @Androser420 also converted many of the attribute selectors to class selectors

using class selectors should improve performance but will make the theme more likely to break on discord updates. with the recent update the theme already looked pretty bad, so at this point i'm not really concerned about that anymore.

**Comment by ninetailedtori** [comment]
Significant improvement using class selectors, it's still a *teeny* bit laggy but it's not that noticeable. Stress testing with panel switching will lag, but I don't honestly think I'll even notice that day to day. It's very good now :)
Typing still has the occasional blip of input lag, but it's not noticeable like before ❤
Thank you for the improvements :)

**Comment by Akasiek** [comment]
Wow! It really does look a lot better. Thank you

**Comment by ninetailedtori** [comment]
Folder opening and closing is super laggy with the folder sidebar plugin. That's my only note currently :3

**Comment by blindmodz** [comment]
At least for me it's more laggy than before

**Comment by refact0r** [comment]
> At least for me it's more laggy than before (tbf i have a lot of plugging installed on vencord)

pretty sure the last discord update just worsened performance in general.

**Comment by itsTyrion** [comment]
now it's mostly the typing for me (Ryzen 5 5600, GTX 1070, Win10), scrolling even absolutely sending it with auto scroll (click mouse wheel and move cursor up)

**Comment by newbee1905** [comment]
I also have lags with typing (Intel i7-9750H, GTX 1650, Linux). Havent tried your midnight theme yet, will check it out if i also have performance issues with it

**Comment by kokoropaint** [comment]
Trying this theme out and the lag has been horrible the second I turn it on. I have a very strong PC and haven't dealt with this with themes before (using Vencord). Tested midnight and it runs beautifully in comparison. Just wanted to report (I imagine the cause of this might be out of your control)
